openeo/rest/datacube.py

```diff
@@ -187,9 +192,14 @@ def load_collection(
                     "Unexpected parameterized `spatial_extent` in `load_collection`:"
                     f" expected schema with type 'object' but got {spatial_extent.schema!r}."
                 )
+        valid_geojson_types = [
+            "Polygon", "MultiPolygon", "Feature", "FeatureCollection"
+        ]
+        if spatial_extent and not (isinstance(spatial_extent, dict) and spatial_extent.keys() & {"west", "east", "north", "south"}):
+            spatial_extent = _get_geometry_argument(argument=spatial_extent,valid_geojson_types=valid_geojson_types,connection=connection)
```

Comment:
This `spatial_extent` handling gets a bit hard to understand,

I'd propose to structure it more as a "switch" construct, like this:
```python
if isinstance(spatial_extent, Parameter):
    ...
elif isinstance(spatial_extent, dict) and spatial_extent.keys() & {"west", "east", "north", "south"}:
    pass
else:
    valid_geojson_types = ["Polygon", "MultiPolygon", "Feature", "FeatureCollection"]
    spatial_extent = _get_geometry_argument(argument=spatial_extent, valid_geojson_types=valid_geojson_types, connection=connection)
```